main.c

```diff
@@ -1067,6 +1062,10 @@ int main(int argc, char *argv[]) {
 			wl_compositor_create_surface(state.compositor);
 	}
 
+	if (!state.cursor_shape_manager && !create_cursors(&state, output)) {
```

Comment:
Moving this right after the second `wl_display_roundtrip()` should be enough: the `wl_output` properties should've been received by then.